src/backend/api/reservations.js

```diff
@@ -2,14 +2,85 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
-router.get("/", async (request, response) => {
+router.get("/res", async (request, response) => {
   try {
     // knex syntax for selecting things. Look up the documentation for knex for further info
     const rows = await knex.raw("select * from Reservation");
-    response.json(rows);
+    response.json(rows[0]);
   } catch (error) {
     throw error;
   }
 });
 
+
+// 1.Returns all reservations
+
+router.get("/reservation", async (request, response) => {
+    try {
+      // knex syntax for selecting things. Look up the documentation for knex for further info
+      const rows = await knex("Reservation").select("*");
+      response.json(rows);
+    } catch (error) {
+      throw error;
+    }
+  });
+   
+// 2.Adds a new reservation  to the database
+  router.post("/", async (req,res) =>{
+    try {
+      const rows = await knex('Reservation').insert({ number_of_guests: 3,meal_id: 1,created_date: '2022-10-11',contact_phonenumber : "22435674",contact_name: "ram",contact_email: "ram@123"});
+     res.json(rows);
```

Comment:
Ofcourse, insert should be based on request body.
And additional comment - may be try to indent code - its much more readable